spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala

```diff
@@ -452,29 +452,36 @@ class DeltaSharingSourceSuite extends QueryTest
       )
       checkAnswer(spark.read.format("parquet").load(outputDir.getCanonicalPath), expected)
 
+      val newQuery = withStreamReaderAtVersion()
+        .option("maxFilesPerTrigger", "1")
+        .load().writeStream.format("console")
+        .option("checkpointLocation", checkpointDir.getCanonicalPath)
+        .start()
+
       // There are 4 checkpoints, remove the latest 2.
       val checkpointFiles = FileUtils.listFiles(checkpointDir, null, true).asScala
       checkpointFiles.foreach{ f =>
         if (!f.isDirectory() &&
-          (f.getCanonicalPath.endsWith("2") || f.getCanonicalPath.endsWith("3"))) {
+          (f.getCanonicalPath.endsWith("2") || f.getCanonicalPath.endsWith("3") ||
+            f.getCanonicalPath.endsWith("2.crc") || f.getCanonicalPath.endsWith("3.crc") ||
+            f.getCanonicalPath.endsWith("metadata.crc"))) {
           f.delete()
         }
+        // SparkStructuredStreaming requires query id match to reuse a checkpoint.
+        if (f.getCanonicalPath.endsWith("metadata")) {
+          FileUtils.writeStringToFile(f, s"""{"id":"${newQuery.id}"}""")
+        }
```

Comment:
Can you please add some comments on what does "*metadata" file contain, and why are we overwriting the existing file?

Reply:
added a comment.